external/llvm-project/mlir/lib/Conversion/AsyncToGPU/AsyncToGPU.cpp

```diff
@@ -284,17 +284,21 @@ class LaunchOpConversion : public OpConversionPattern<async::LaunchOp> {
 //===----------------------------------------------------------------------===//
 
 namespace {
-class AwaitOpConversion : public OpConversionPattern<async::AwaitOp> {
-public:
-  using OpConversionPattern::OpConversionPattern;
+struct AwaitRewritePattern : public OpRewritePattern<async::AwaitOp> {
+  using OpRewritePattern<async::AwaitOp>::OpRewritePattern;
 
-  LogicalResult matchAndRewrite(async::AwaitOp op, OpAdaptor adaptor,
-                                ConversionPatternRewriter &rw) const override {
+  LogicalResult matchAndRewrite(async::AwaitOp op,
+                                PatternRewriter &rw) const override {
     auto tokenType = rw.getType<gpu::AsyncTokenType>();
-    rw.create<gpu::WaitOp>(op.getLoc(), tokenType, adaptor.getOperands());
-    rw.eraseOp(op);
+    Value input = op->getOperand(0);
+    if (input.getType() == tokenType) {
+      assert(op.getResultType() == None);
```

Comment:
What's this assert for?

Reply:
async.await should never has a result when the operand is a token